sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -74,12 +75,26 @@ class QueryExecution(
     sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
   }
 
+  lazy val nonRootCommandExecuted: LogicalPlan = analyzed mapChildren { child =>
+    child transformDown {
+      // SPARK-35378: Eagerly execute non-root Command
```

Comment:
Instead of just a JIRA number, as this is not TODO item, can you simply add a few comment here explaining why you need to eagerly execute `Command` under top plan node?

Reply:
OK